src/ResourceManager/Network/Commands.Network/NetworkWatcher/ConnectionMonitor/NewAzureNetworkWatcherConnectionMonitorCommand.cs

```diff
@@ -0,0 +1,224 @@
+            HelpMessage = "The name of network watcher.",
+            ParameterSetName = "SetByName")]
+        [ValidateNotNullOrEmpty]
+        public string NetworkWatcherName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the network watcher resource group.",
+            ParameterSetName = "SetByName")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Location of the network watcher.",
+            ParameterSetName = "SetByLocation")]
+        [LocationCompleter("Microsoft.Network/networkWatchers/connectionMonitors")]
+        [ValidateNotNull]
+        public string Location { get; set; }
+
+        [Alias("ConnectionMonitorName")]
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The connection monitor name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The ID of the connection monitor source.")]
+        [ValidateNotNullOrEmpty]
+        public string SourceResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Monitoring interval in seconds.")]
+        [ValidateNotNullOrEmpty]
+        public int? MonitoringIntervalInSeconds { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Source port.")]
+        [ValidateNotNull]
+        [ValidateRange(1, int.MaxValue)]
+        public int SourcePort { get; set; }
+
+        [Parameter(
+             Mandatory = false,
+             HelpMessage = "The ID of the connection monitor destination.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The Ip address of the connection monitor destination.")]
+        [ValidateNotNullOrEmpty]
+        public string DestinationAddress { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Destination port.")]
+        [ValidateNotNull]
+        [ValidateRange(1, int.MaxValue)]
+        public int DestinationPort { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Configure connection monitor, but do not start it")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter ConfigureOnly { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "A hashtable which represents resource tags.")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Do not ask for confirmation if you want to overwrite a resource")]
+        public SwitchParameter Force { get; set; }
```

Comment:
@irrogozh is this `Force` parameter necessary? We reserve the parameter for when we want the cmdlet to prompt the user for additional confirmation of the action they are about to perform.

Reply:
Yes, we want customer to confirm if he really wants to overwrite resource

Reply:
@irrogozh just to confirm, if connection monitor "Foo" exists, and you try to create another one called "Foo" in the same network watcher, it overwrites it instead of failing?